cmd/bosun/expr/parse/parse.go

```diff
@@ -31,13 +31,14 @@ type Tree struct {
 }
 
 type Func struct {
-	Args     []models.FuncType
-	Return   models.FuncType
-	Tags     func([]Node) (Tags, error)
-	F        interface{}
-	VArgs    bool
-	VArgsPos int
-	Check    func(*Tree, *FuncNode) error
+	Args      []models.FuncType
+	Return    models.FuncType
+	Tags      func([]Node) (Tags, error)
+	F         interface{}
+	VArgs     bool
+	VArgsPos  int
+	VArgsOmit bool
```

Comment:
Most of the time it isn't true. So when defining funcs you don't have to
include it. So false is a better default

On Mon, Mar 14, 2016 at 1:19 PM, Craig Peterson <notifications@github.com>
wrote:

> In cmd/bosun/expr/parse/parse.go
> <https://github.com/bosun-monitor/bosun/pull/1667#discussion_r56038601>:
>
> > @@ -31,13 +31,14 @@ type Tree struct {
> >  }
> >
> >  type Func struct {
> > -	Args     []models.FuncType
> > -	Return   models.FuncType
> > -	Tags     func([]Node) (Tags, error)
> > -	F        interface{}
> > -	VArgs    bool
> > -	VArgsPos int
> > -	Check    func(*Tree, *FuncNode) error
> > +	Args      []models.FuncType
> > +	Return    models.FuncType
> > +	Tags      func([]Node) (Tags, error)
> > +	F         interface{}
> > +	VArgs     bool
> > +	VArgsPos  int
> > +	VArgsOmit bool
>
> How about inverting it. Maybe VArgsRequired or even MinVArgs would be
> clearer.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bosun-monitor/bosun/pull/1667/files#r56038601>.
>